test/functional/aws_test.go

```diff
@@ -0,0 +1,162 @@
+package functional
+
+import (
+	goctx "context"
+	"errors"
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/elasticache"
+	"github.com/aws/aws-sdk-go/service/rds"
+	"github.com/aws/aws-sdk-go/service/s3"
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"os"
+	"testing"
+)
+
+//TODO update when Fuse Online is ready as per https://gitlab.cee.redhat.com/integreatly-qe/integreatly-test-cases/blob/master/tests/high-availability/f01-verify-all-aws-resources-are-created-in-aws.md
+const (
+	RDSSuffix                  = "postgres-integreatly"
+	UPSResourceName            = "ups-" + RDSSuffix
+	CRResourceName             = "codeready-" + RDSSuffix
+	ClusterSSOResourceName     = "rhsso-" + RDSSuffix
+	ApplicationSSOResourceName = "rhssouser-" + RDSSuffix
+	ThreeScaleResourceName     = "threescale" + RDSSuffix
+	ThreeScaleNameElement      = "rhmioperatorthre"
+	S3Backup                   = "rhmioperatorback"
+	resourceStatusAvailable    = "available"
+	awsCredsNamespace          = "kube-system"
+	awsCredsSecretName         = "aws-creds"
+)
+
+var (
+	expectedResources = []string{
+		ThreeScaleResourceName,
+		UPSResourceName,
+		CRResourceName,
+		ClusterSSOResourceName,
+		ApplicationSSOResourceName,
+	}
+)
+
+func TestAWS(t *testing.T) {
+	//TODO
+}
+
+func testExpectedResourcesExist(t testing.T, f *framework.Framework, instances []*rds.DBInstance, cacheClusters []*elasticache.CacheCluster, buckets []*s3.Bucket, expectedResources []string) (map[string]bool, error) {
```

Comment:
Does this function have to be wrapped? Could probably just have this as the main function

Reply:
Changed to be used as the main function :+1: